# World Crime Rate and Alcohol Consumption
## Levi Nickerson
## Link to description video: https://www.youtube.com/watch?v=ytdrljCIlvc&feature=youtu.be

This report is an examination of alcohol consumption around the globe and its potential effects on crime rates. The primary goal of this report was to identify any correlations that may be drawn between alcohol consumption and criminal activity. The project consisted of merging a large amount of data into a created database. The data consisted of multiple datasets that each had to cleaned and prepared for analysis. The database was then hosted on elephantsql, accessed using sqlalchemy, updated with the data, and queried. In order to determine correlations, four different groups of queries were computed. The first group of queries was centered on criminal activity. The rates were queried in an attempt to identify criminal activity that could predict alcohol consumption. The second group of queries focused on alcohol consumption. The queries were used to determine if alcohol consumption could serve as an accurate predictor for criminal activity. The third group of queries focused on the types of alcohol consumed and their potential impact on criminal activity. The final group of queries centered on the United States. Queries were used to attempt to predict the crime rates within the United States.

# Alcohol Dataset

As mentioned above this project utilized several data sets. The first of which was associated with alcohol consumption for 193 countries worldwide. The dataset holds data that was collected from the year 2010. It consists of the number of servings per person for the year for beer, spirits, and wine. It also includes data for the total liters consumed per person in the year 2010 (this value has been converted into gallons for this report). The dataset was clean and required no maintenance. The only potential errors in the dataset could have occurred during data collection. For example, Afghanistan recorded to have 0 for all categories. This is due to the fact that it is illegal to consume alcohol in Afghanistan and thus near impossible to estimate how much drinking actually goes on. Other than specific cases like this one, the dataset should be an accurate estimation for the year 2010.

# Alcohol Plots
The following plots were generated using the plotly website. The plots provide a visualization for the alcohol dataset. The most important plot is the final plot that totals all of the alcohol consumed per person. This plot provides an important estimate to highlight which countries consume the most alcohol. These plots alone do not provide everything necessary, but provide an interesting view.

Note: White and Gray countries correspond to not present or null data

<img src ='ImagesProject/BeerServingsPerPerson2010.png'>
<img src='ImagesProject/SpiritsServingsPerPerson2010.png'>
<img src='ImagesProject/WineServingsPerPerson2010.png'>
<img src='ImagesProject/TotalServingsPerPerson2010.png'>

# Crime Datasets
The remaining data sets used for the project all regarded crime rates. The crime rate for each data set is the __number of instances per 100,000 population__. The data sets cover a wide variety of crimes that are described below. The definitions listed below were taken directly from the datasets. These datasets required a little bit more effort to clean them. The datasets all consist of crimes for the years 2000 to approximately 2012. The reason the data needed to be processed was because data was missing for the year 2010 for some countries. For some of the datasets a good number of the countries had data for 2010 and others did not. In order to attempt to fill in for missing values I took the average of the previous three years if the data value was missing. This helped to provide an estimate for missing values and provide a more in depth study. Some of the datasets were somewhat sparse. The only explanation I can attribute to this is that some governments do not have the time or money to record this data. For example, African countries had very few entries in any of these data sets. Africa is the poorest continent and it would make sense that African governments have more important problems than recording full datasets. 

# Definitions
Assault: physical attack against the body of another person resulting in serious bodily injury; excluding indecent/sexual assault; threats and slapping/punching. 'Assault' leading to death should also be excluded.

Burglary: gaining unauthorized access to a part of a building/dwelling or other premises; including by use of force; with the intent to steal goods (breaking and entering). “Burglary” should include; where possible; theft from a house; apartment or other dwelling place; factory; shop or office; from a military establishment; or by using false keys. It should exclude theft from a car; from a container; from a vending machine; from a parking meter and from fenced meadow/compound.

Domestic Burglary: theft from a house, apartment or other dwelling place

Homicide: unlawful death purposefully inflicted on a person by another person

Motor Vehicle Theft: removal of a motor vehicle without the consent of the owner of the vehicle. 'Motor Vehicles' includes all land vehicles with an engine that run on the road, including cars, motorcycles, buses, lorries, construction and agricultural vehicles

Robbery: theft of property from a person; overcoming resistance by force or threat of force. Where possible, the category “Robbery” should include muggings (bag-snatching) and theft with violence; but should exclude pick pocketing and extortion

Sexual Violence (Minors):  crimes of sexual nature committed against minors of age as defined

Sexual Violence (All): rape and sexual assault; including sexual offences against children





# Crime Plots
The following plots were also generated using the plotly website. The plots provide a very important visualization for the entire project. These plots are important because they highlight exactly which countries are missing data. For example, it should be fairly obvious that there is not crime data for the United States. I elected to leave the US data out to determine if I could develop an accurate prediction for the crime rates in the US. At the end of the report I will attempt to make this prediction and fully explain where the US lines up with the rest of the world. The visualizations also highlight the real world difficulties of maintaining data sets such as these. Many countries do not have the time or resources to properly collect this kind of data.  

Note: White and Gray countries correspond to not present or null data

<img src='ImagesProject/AssaultRatePer100,000Population.png'>
<img src='ImagesProject/BurglaryRatePer100,000Population.png'>
<img src='ImagesProject/DomesticBurglaryRatePer100,000Population.png'>
<img src='ImagesProject/HomicideRatePer100,000Population.png'>
<img src='ImagesProject/MotorVehicleTheftRatePer100,000Population.png'>
<img src='ImagesProject/RobberyRatePer100,000Population.png'>
<img src='ImagesProject/SexualViolenceMinorsRatePer100,000Population.png'>
<img src='ImagesProject/SexualViolenceAllRatePer100,000Population.png'>

# Database Creation

The following code segments show exactly how I created the database. To start the whole project I processed and combined all of the datasets using Python. Once I had a unified dataset, I created an instance on ElephantSQL. I accessed my database with sqlalchemy raw and added all of the data into the ElaphantSQL instance. The remainder of the project will consist of queries of the database using sqlalchemy.

In [1]:
from sqlalchemy import create_engine
import sqlalchemy.sql

In [2]:
passwd ='BoiUdeP29RL1PesOEb76e4fO7f973W5L'
dbuser='leni2025@colorado.edu'
eng = create_engine('postgresql://vmercdoa:BoiUdeP29RL1PesOEb76e4fO7f973W5L@otto.db.elephantsql.com:5432/vmercdoa'.format(dbuser, passwd))
con = eng.connect()

In [3]:
rs = con.execute('''
DROP TABLE IF EXISTS data;
CREATE TABLE data
    (
      country character varying(50),
      beer float,
      spirit float,
      wine float,
      pure float,
      assault float,
      burglary float,
      burglary_domestic float,
      homicide float,
      motor_vehicle_theft float,
      robbery float,
      sexual_violence_minors float,
      sexual_violence_all float
    );
    
''')

In [4]:
import pandas as pd

data = pd.read_csv('total.csv')
data = data.to_dict('record')

In [5]:
for row in data:
    cmd = sqlalchemy.sql.text('''INSERT INTO data(country, beer, spirit, wine, pure, assault, burglary, burglary_domestic, homicide, motor_vehicle_theft, robbery, sexual_violence_minors, sexual_violence_all)\
    VALUES ( :country, :beer_servings, :spirit_servings, :wine_servings, :total_litres_of_pure_alcohol, :assault, :burglary, :burglary_domestic, :homicide, :motor_vehicle_theft, :robbery, :sexual_violence_minors, :sexual_violence_all)''')
    con.execute(cmd, row)

# Aggregate Queries

I first started by computing aggregate values for each column of the dataset in order to get a general view of the data. The aggregates are listed below in two tables. The aggregates will then be used in various queries throughout the rest of the project.

In [6]:
average = con.execute('''SELECT avg(NULLIF(beer,0)), avg(NULLIF(spirit,0)), avg(NULLIF(wine,0)), avg(NULLIF(pure,0))
               FROM data
               ''').fetchall()

beer_avg = round(float(average[0][0]),2)
spirit_avg = round(float(average[0][1]),2)
wine_avg = round(float(average[0][2]),2)
total_avg = round(float(average[0][3]),2)

In [7]:
average_crime = con.execute('''SELECT avg(NULLIF(assault,0)), avg(NULLIF(burglary,0)), avg(NULLIF(burglary_domestic,0)), avg(NULLIF(homicide,0)),
                               avg(NULLIF(motor_vehicle_theft,0)), avg(NULLIF(robbery,0)), avg(NULLIF(sexual_violence_minors,0)), avg(NULLIF(sexual_violence_all,0))
                               FROM data
                               ''').fetchall()

assault_avg = round(float(average_crime[0][0]),2)
burglary_avg = round(float(average_crime[0][1]),2)
burglary_domestic_avg = round(float(average_crime[0][2]),2)
homicide_avg = round(float(average_crime[0][3]),2)
motor_vehicle_theft_avg = round(float(average_crime[0][4]),2)
robbery_avg = round(float(average_crime[0][5]),2)
sexual_violence_minors_avg = round(float(average_crime[0][6]),2)
sexual_violence_all_avg = round(float(average_crime[0][7]),2)

In [8]:
from IPython.display import HTML, display
display(HTML("<table>  <tr> <th> Alcohol Consumed </th> <th> Average Number of Servings </th> </tr> \
                       <tr> <td> Beer </td> <td> {beer_avg} </td></tr> \
                       <tr> <td> Spirits </td> <td> {spirit_avg} </td></tr> \
                       <tr> <td> Wine </td> <td> {wine_avg} </td></tr> \
                         </table>".format(beer_avg = beer_avg, spirit_avg = spirit_avg, wine_avg = wine_avg)))

print("Average Amount of Pure Alcohol Consumed Per Person: ", total_avg, "Gallons")
print("")
display(HTML("<table>  <tr> <th> Country </th> <th> Average Crime Rate </th> </tr> \
                       <tr> <td> Assault </td> <td> {assault_avg} </td></tr> \
                       <tr> <td> Burglary </td> <td> {burglary_avg} </td></tr> \
                       <tr> <td> Burglary Domestic </td> <td> {burglary_domestic_avg} </td></tr> \
                       <tr> <td> Homicide </td> <td> {homicide_avg} </td></tr> \
                       <tr> <td> Motor Vehicle Theft </td> <td> {motor_vehicle_theft_avg} </td></tr> \
                       <tr> <td> Robbery </td> <td> {robbery_avg} </td></tr> \
                       <tr> <td> Sexual Violence (Minors) </td> <td> {sexual_violence_minors_avg} </td></tr> \
                       <tr> <td> Sexual Violence (All) </td> <td> {sexual_violence_all_avg} </td></tr> \
                       </table>".format(assault_avg = assault_avg, burglary_avg = burglary_avg, burglary_domestic_avg = burglary_domestic_avg, homicide_avg = homicide_avg, motor_vehicle_theft_avg = motor_vehicle_theft_avg,
                                        robbery_avg = robbery_avg, sexual_violence_minors_avg = sexual_violence_minors_avg, sexual_violence_all_avg = sexual_violence_all_avg)))
print("A reminder for the crime rate, the rate is instances per 100,000 population.")
print("")

Average Amount of Pure Alcohol Consumed Per Person:  1.34 Gallons



A reminder for the crime rate, the rate is instances per 100,000 population.



These tables provided a couple of interesting insights. Unsurprisingly, beer was the most common alcoholic beverage worldwide in 2010. The average person consumed 1.34 gallons of pure alcohol in 2010. That is quite a large amount considering that each serving of alcohol only consists of 0.6 ounces. From the crime table you can see that burglary had the highest rate and homicide had the lowest rate in 2010. 

I performed another aggregation query in order to get a better picture of the skew of the criminal datasets. I performed a query to identify how much the top ten offenders in each criminal category contributed to the average rate. The results are summarized in a table below.

In [9]:
crime_totals = con.execute('''SELECT sum(NULLIF(assault, 0)), sum(NULLIF(burglary,0)), sum(NULLIF(burglary_domestic,0)), sum(NULLIF(homicide,0)), sum(NULLIF(motor_vehicle_theft,0)), sum(NULLIF(robbery,0)),
                              sum(NULLIF(sexual_violence_minors, 0)), sum(NULLIF(sexual_violence_all, 0))
                              FROM DATA
                              ''').fetchall()
assault_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(assault,0)) as total
                             FROM data
                             where (assault >= {assault_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(assault_avg = assault_avg)).fetchall()
burglary_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(burglary,0)) as total
                             FROM data
                             where (burglary >= {burglary_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(burglary_avg = burglary_avg)).fetchall()
burglary_domestic_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(burglary_domestic,0)) as total
                             FROM data
                             where (burglary_domestic >= {burglary_domestic_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(burglary_domestic_avg = burglary_domestic_avg)).fetchall()
homicide_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(homicide,0)) as total
                             FROM data
                             where (homicide >= {homicide_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(homicide_avg = homicide_avg)).fetchall()
motor_vehicle_theft_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(motor_vehicle_theft,0)) as total
                             FROM data
                             where (motor_vehicle_theft >= {motor_vehicle_theft_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(motor_vehicle_theft_avg = motor_vehicle_theft_avg)).fetchall()
robbery_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(robbery,0)) as total
                             FROM data
                             where (robbery >= {robbery_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(robbery_avg = robbery_avg)).fetchall()
sexual_violence_minors_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(sexual_violence_minors,0)) as total
                             FROM data
                             where (sexual_violence_minors >= {sexual_violence_minors_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(sexual_violence_minors_avg = sexual_violence_minors_avg)).fetchall()
sexual_violence_all_ten = con.execute('''SELECT sum(old.total)
                             FROM (SELECT country, sum(NULLIF(sexual_violence_all,0)) as total
                             FROM data
                             where (sexual_violence_all >= {sexual_violence_all_avg})
                             GROUP BY country
                             ORDER BY total desc
                             LIMIT 10) as old'''
                             .format(sexual_violence_all_avg = sexual_violence_all_avg)).fetchall()

assault_ten_cont = round(assault_ten[0][0]/crime_totals[0][0],2)
burglary_ten_cont = round(burglary_ten[0][0]/crime_totals[0][1],2)
burglary_domestic_ten_cont = round(burglary_domestic_ten[0][0]/crime_totals[0][2],2)
homicide_ten_cont = round(homicide_ten[0][0]/crime_totals[0][3],2)
motor_vehicle_theft_ten_cont = round(motor_vehicle_theft_ten[0][0]/crime_totals[0][4],2)
robbery_ten_cont = round(robbery_ten[0][0]/crime_totals[0][5],2)
sexual_violence_minors_ten_cont = round(sexual_violence_minors_ten[0][0]/crime_totals[0][6],2)
sexual_violence_all_ten_cont = round(sexual_violence_all_ten[0][0]/crime_totals[0][7],2)

display(HTML("<table>  <tr> <th> Crime </th>                          <th> Top Ten Contribution to Total </th> </tr> \
                       <tr> <td> Assault </td>                        <td> {assault_ten_cont} </td>  </tr> \
                       <tr> <td> Burglary </td>                       <td> {burglary_ten_cont} </td> </tr> \
                       <tr> <td> Burglary Domestic</td>               <td> {burglary_domestic_ten_cont} </td> </tr> \
                       <tr> <td> Homicde </td>                        <td> {homicide_ten_cont} </td> </tr> \
                       <tr> <td> Motor Vehicle Theft </td>            <td> {motor_vehicle_theft_ten_cont} </td> </tr> \
                       <tr> <td> Robbery </td>                        <td> {robbery_ten_cont} </td> </tr> \
                       <tr> <td> Sexual Violence (Minors) </td>       <td> {sexual_violence_minors_ten_cont} </td> </tr> \
                       <tr> <td> Sexual Violence (All) </td>          <td> {sexual_violence_all_ten_cont} </td> </tr> \
                       </table>".format(assault_ten_cont = assault_ten_cont, burglary_ten_cont = burglary_ten_cont, burglary_domestic_ten_cont = burglary_domestic_ten_cont, homicide_ten_cont = homicide_ten_cont,
                                        motor_vehicle_theft_ten_cont = motor_vehicle_theft_ten_cont, robbery_ten_cont = robbery_ten_cont, sexual_violence_minors_ten_cont = sexual_violence_minors_ten_cont,
                                        sexual_violence_all_ten_cont = sexual_violence_all_ten_cont)))

This table highlights a very important issue. For each crime rate the top ten countries contribute nearly half. This fact highlights just how dangerous certain countries are. A small select group of countries can greatly raise the average rate of crimes worldwide. 

# Criminal Rates as a Predictor for Alcohol Consumption.

I wanted to start with using crime rate to predict if a country has a high alcohol consumption. In other words, if a country has high crime rate, how likely is it that the country has an above average alcohol consumption. I performed the calculation for each of the crimes and the results are listed below in a table. 

In [10]:
# Assault
number_assault_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and assault >= {assault_avg})'''.format(pure_avg = total_avg, assault_avg = assault_avg)).fetchall()
number_assault = con.execute('''SELECT count(*) number
                                FROM data
                                where (assault >= {assault_avg})'''.format(assault_avg = assault_avg)).fetchall()
assault_total = con.execute('''SELECT count(NULLIF(assault,0)) number
                               FROM data''').fetchall()
assault_odds1 = round(number_assault[0][0]/assault_total[0][0],2)
assault_odds2 = round(number_assault_drink[0][0]/number_assault[0][0],2)

# Burglary
number_burglary_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and burglary >= {burglary_avg})'''.format(pure_avg = total_avg, burglary_avg = burglary_avg)).fetchall()
number_burglary = con.execute('''SELECT count(*) number
                                FROM data
                                where (burglary >= {burglary_avg})'''.format(burglary_avg = burglary_avg)).fetchall()
burglary_total = con.execute('''SELECT count(NULLIF(burglary,0)) number
                               FROM data''').fetchall()
burglary_odds1 = round(number_burglary[0][0]/burglary_total[0][0],2)
burglary_odds2 = round(number_burglary_drink[0][0]/number_burglary[0][0],2)

# Burglary domestic
number_burglary_domestic_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and burglary_domestic >= {burglary_domestic_avg})'''.format(pure_avg = total_avg, burglary_domestic_avg = burglary_domestic_avg)).fetchall()
number_burglary_domestic = con.execute('''SELECT count(*) number
                                FROM data
                                where (burglary_domestic >= {burglary_domestic_avg})'''.format(burglary_domestic_avg = burglary_domestic_avg)).fetchall()
burglary_domestic_total = con.execute('''SELECT count(NULLIF(burglary_domestic,0)) number
                               FROM data''').fetchall()
burglary_domestic_odds1 = round(number_burglary_domestic[0][0]/burglary_domestic_total[0][0],2)
burglary_domestic_odds2 = round(number_burglary_domestic_drink[0][0]/number_burglary_domestic[0][0],2)

# Homicide
number_homicide_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and homicide >= {homicide_avg})'''.format(pure_avg = total_avg, homicide_avg = homicide_avg)).fetchall()
number_homicide = con.execute('''SELECT count(*) number
                                FROM data
                                where (homicide >= {homicide_avg})'''.format(homicide_avg = homicide_avg)).fetchall()
homicide_total = con.execute('''SELECT count(NULLIF(homicide,0)) number
                               FROM data''').fetchall()
homicide_odds1 = round(number_homicide[0][0]/homicide_total[0][0],2)
homicide_odds2 = round(number_homicide_drink[0][0]/number_homicide[0][0],2)

# Motor Vehicle Theft
number_motor_vehicle_theft_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and motor_vehicle_theft >= {motor_vehicle_theft_avg})'''.format(pure_avg = total_avg, motor_vehicle_theft_avg = motor_vehicle_theft_avg)).fetchall()
number_motor_vehicle_theft = con.execute('''SELECT count(*) number
                                FROM data
                                where (motor_vehicle_theft >= {motor_vehicle_theft_avg})'''.format(motor_vehicle_theft_avg = motor_vehicle_theft_avg)).fetchall()
motor_vehicle_theft_total = con.execute('''SELECT count(NULLIF(motor_vehicle_theft,0)) number
                               FROM data''').fetchall()
motor_vehicle_theft_odds1 = round(number_motor_vehicle_theft[0][0]/motor_vehicle_theft_total[0][0],2)
motor_vehicle_theft_odds2 = round(number_motor_vehicle_theft_drink[0][0]/number_motor_vehicle_theft[0][0],2)

# Robbery
number_robbery_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and robbery >= {robbery_avg})'''.format(pure_avg = total_avg, robbery_avg = robbery_avg)).fetchall()
number_robbery = con.execute('''SELECT count(*) number
                                FROM data
                                where (robbery >= {robbery_avg})'''.format(robbery_avg = robbery_avg)).fetchall()
robbery_total = con.execute('''SELECT count(NULLIF(robbery,0)) number
                               FROM data''').fetchall()
robbery_odds1 = round(number_robbery[0][0]/robbery_total[0][0],2)
robbery_odds2 = round(number_robbery_drink[0][0]/number_robbery[0][0],2)

# Sexual Violence Minors
number_sexual_violence_minors_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and sexual_violence_minors >= {sexual_violence_minors_avg})'''.format(pure_avg = total_avg, sexual_violence_minors_avg = sexual_violence_minors_avg)).fetchall()
number_sexual_violence_minors = con.execute('''SELECT count(*) number
                                FROM data
                                where (sexual_violence_minors >= {sexual_violence_minors_avg})'''.format(sexual_violence_minors_avg = sexual_violence_minors_avg)).fetchall()
sexual_violence_minors_total = con.execute('''SELECT count(NULLIF(sexual_violence_minors,0)) number
                               FROM data''').fetchall()
sexual_violence_minors_odds1 = round(number_sexual_violence_minors[0][0]/sexual_violence_minors_total[0][0],2)
sexual_violence_minors_odds2 = round(number_sexual_violence_minors_drink[0][0]/number_sexual_violence_minors[0][0],2)

# Sexual Violence All
number_sexual_violence_all_drink = con.execute('''SELECT count(*) number
                                      FROM data
                                      where (pure >= {pure_avg} and sexual_violence_all >= {sexual_violence_all_avg})'''.format(pure_avg = total_avg, sexual_violence_all_avg = sexual_violence_all_avg)).fetchall()
number_sexual_violence_all = con.execute('''SELECT count(*) number
                                FROM data
                                where (sexual_violence_all >= {sexual_violence_all_avg})'''.format(sexual_violence_all_avg = sexual_violence_all_avg)).fetchall()
sexual_violence_all_total = con.execute('''SELECT count(NULLIF(sexual_violence_all,0)) number
                               FROM data''').fetchall()
sexual_violence_all_odds1 = round(number_sexual_violence_all[0][0]/sexual_violence_all_total[0][0],2)
sexual_violence_all_odds2 = round(number_sexual_violence_all_drink[0][0]/number_sexual_violence_all[0][0],2)


# Table
display(HTML("<table>  <tr> <th> Crime </th>                          <th> Odds of At or Above Average Crime Rate </th>         <th> Odds of At or Above Average Alcohol Consumption Given Above Average Crime Rate </th> </tr> \
                       <tr> <td> Assault </td>                        <td> {assault_odds1} </td>                                <td> {assault_odds2} </td> </tr> \
                       <tr> <td> Burglary </td>                       <td> {burglary_odds1} </td>                               <td> {burglary_odds2} </td> </tr> \
                       <tr> <td> Burglary Domestic</td>               <td> {burglary_domestic_odds1} </td>                      <td> {burglary_domestic_odds2} </td> </tr> \
                       <tr> <td> Homicde </td>                        <td> {homicide_odds1} </td>                               <td> {homicide_odds2} </td> </tr> \
                       <tr> <td> Motor Vehicle Theft </td>            <td> {motor_vehicle_theft_odds1} </td>                    <td> {motor_vehicle_theft_odds2} </td> </tr> \
                       <tr> <td> Robbery </td>                        <td> {robbery_odds1} </td>                                <td> {robbery_odds2} </td> </tr> \
                       <tr> <td> Sexual Violence (Minors) </td>       <td> {sexual_violence_minors_odds1} </td>                 <td> {sexual_violence_minors_odds2} </td> </tr> \
                       <tr> <td> Sexual Violence (All) </td>          <td> {sexual_violence_all_odds1} </td>                    <td> {sexual_violence_all_odds2} </td> </tr> \
              </table>".format(assault_odds1 = assault_odds1, assault_odds2 = assault_odds2, burglary_odds1 = burglary_odds1, burglary_odds2 = burglary_odds2, burglary_domestic_odds1 = burglary_domestic_odds1, 
                               burglary_domestic_odds2 = burglary_domestic_odds2, homicide_odds1 = homicide_odds1, homicide_odds2 = homicide_odds2, motor_vehicle_theft_odds1 = motor_vehicle_theft_odds1,
                               motor_vehicle_theft_odds2 = motor_vehicle_theft_odds2, robbery_odds1 = robbery_odds1, robbery_odds2 = robbery_odds2, sexual_violence_minors_odds1 = sexual_violence_minors_odds1,
                               sexual_violence_minors_odds2 = sexual_violence_minors_odds2, sexual_violence_all_odds1 = sexual_violence_all_odds1, sexual_violence_all_odds2 = sexual_violence_all_odds2)))

alcohol_total_number = con.execute('''SELECT count(NULLIF(pure, 0))
                           FROM data''').fetchall()
alcohol_over_number = con.execute('''SELECT count(NULLIF(pure, 0))
                          FROM data
                          WHERE pure >= {total_avg}'''.format(total_avg = total_avg)).fetchall()
alcohol_odds = round(alcohol_over_number[0][0]/alcohol_total_number[0][0],2)
print("The odds of a country having an above average alcohol consumption are: ", alcohol_odds)
print("")
print("Sample Meaning for the Table: A country has a 31% chance of having an above average assault rate. If that country has an above average assault rate it has a 73% chance of having an above average alcohol consumption.")
                        
    



The odds of a country having an above average alcohol consumption are:  0.47

Sample Meaning for the Table: A country has a 31% chance of having an above average assault rate. If that country has an above average assault rate it has a 73% chance of having an above average alcohol consumption.


This query produced some very interesting results. In general it seems that if a country has a high crime right it is very likely that it also has an above average alcohol consumption. The major generalization that can be made is that if a country has an above average crime rate in any category (except for homicide) there is at least a 70% chance that the country has an above average alcohol consumption. This query shows a positive correlation between criminal activity and alcohol consumption. In other words, criminal activity is a measure that can be used to predict alcohol consumption. The most surprising insight I found from this table was the odds for homicide. A high homicide rate actually suggests a below average alcohol consumption. A potential explanation for this idea is that humans may be more likely to make poor decisions while intoxicated, but they do not necessarily make the worst decisions possible. Another surprising insight I found was the assault odds. Generally intoxicated people are associated with being loud and overly aggressive. I was surprised to see that there is not a direct correlation between assault rates and alcohol consumptoin. Overall, this group of queries highlights the correlation between crime rate predicting alcohol consumption.     

# Alcohol Consumption as a Predictor for Criminal Activity.

Crime rate was found to be a good predictor for criminal activity. The reverse of this relationship was studied below. Can alcohol consumption serve as a predictor for criminal activity?  

In [11]:
alcohol_assault_odds = round(number_assault_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_burglary_odds = round(number_burglary_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_burglary_domestic_odds = round(number_burglary_domestic_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_homicide_odds = round(number_homicide_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_motor_vehicle_theft_odds = round(number_motor_vehicle_theft_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_robbery_odds = round(number_robbery_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_sexual_violence_minors_odds = round(number_sexual_violence_minors_drink[0][0]/alcohol_over_number[0][0],2)
alcohol_sexual_violence_all_odds = round(number_sexual_violence_all_drink[0][0]/alcohol_over_number[0][0],2)

display(HTML("<table>  <tr> <th> Crime </th>                          <th> Odds of Above Average Crime Rate Given Above Average Alcohol Consumption </th> </tr> \
                       <tr> <td> Assault </td>                        <td> {alcohol_assault_odds} </td>  </tr> \
                       <tr> <td> Burglary </td>                       <td> {alcohol_burglary_odds} </td> </tr> \
                       <tr> <td> Burglary Domestic</td>               <td> {alcohol_burglary_domestic_odds} </td> </tr> \
                       <tr> <td> Homicde </td>                        <td> {alcohol_homicide_odds} </td> </tr> \
                       <tr> <td> Motor Vehicle Theft </td>            <td> {alcohol_motor_vehicle_theft_odds} </td> </tr> \
                       <tr> <td> Robbery </td>                        <td> {alcohol_robbery_odds} </td> </tr> \
                       <tr> <td> Sexual Violence (Minors) </td>       <td> {alcohol_sexual_violence_minors_odds} </td> </tr> \
                       <tr> <td> Sexual Violence (All) </td>          <td> {alcohol_sexual_violence_all_odds} </td> </tr> \
                       </table>".format(alcohol_assault_odds = alcohol_assault_odds, alcohol_burglary_odds = alcohol_burglary_odds, alcohol_burglary_domestic_odds = alcohol_burglary_domestic_odds, alcohol_homicide_odds = alcohol_homicide_odds,
                                        alcohol_motor_vehicle_theft_odds = alcohol_motor_vehicle_theft_odds, alcohol_robbery_odds = alcohol_robbery_odds, alcohol_sexual_violence_minors_odds = alcohol_sexual_violence_minors_odds,
                                        alcohol_sexual_violence_all_odds = alcohol_sexual_violence_all_odds)))

The results of these queries were very surprising to me. The results show that alcohol consumption is not a very good predictor for criminal activity. The major generalization from the table is that if a country consumes alcohol at an above average rate then there is at most a 33% chance that the country has an above average rate of crime in any category. This was a very surprising discovery to me. The only possible explanation I could muster for this discovery was the idea that a country that drinks more has more drinking experience and thus might be able to behave more appropriately. The key take away from these queries is that it is more than possible for a country to consume a large amount of alcohol and not end up involved in an above average amount of criminal activity. 

# Types of Alcohol and Their Impact on Criminal Activity

These queries were intended to investigate if there are any correlations between different types of alcohol and criminal activity. The queries computed the criminal rates for countries that above average rates of consumption for beer, spirits, and wine. The results are presented in a table below.

In [12]:
beer_crime_ave = con.execute('''SELECT sum(NULLIF(assault, 0))/count(NULLIF(assault, 0)), sum(NULLIF(burglary, 0))/count(NULLIF(burglary, 0)), sum(NULLIF(burglary_domestic, 0))/count(NULLIF(burglary_domestic, 0)), 
                                       sum(NULLIF(homicide, 0))/count(NULLIF(homicide, 0)), sum(NULLIF(motor_vehicle_theft, 0))/count(NULLIF(motor_vehicle_theft, 0)), sum(NULLIF(robbery, 0))/count(NULLIF(robbery, 0)), 
                                       sum(NULLIF(sexual_violence_minors, 0))/count(NULLIF(sexual_violence_minors, 0)), sum(NULLIF(sexual_violence_all, 0))/count(NULLIF(sexual_violence_all, 0))
                                       FROM data
                                       WHERE beer >= {beer_avg}'''.format(beer_avg = beer_avg)).fetchall()
bca = beer_crime_ave
spirit_crime_ave = con.execute('''SELECT sum(NULLIF(assault, 0))/count(NULLIF(assault, 0)), sum(NULLIF(burglary, 0))/count(NULLIF(burglary, 0)), sum(NULLIF(burglary_domestic, 0))/count(NULLIF(burglary_domestic, 0)), 
                                         sum(NULLIF(homicide, 0))/count(NULLIF(homicide, 0)), sum(NULLIF(motor_vehicle_theft, 0))/count(NULLIF(motor_vehicle_theft, 0)), sum(NULLIF(robbery, 0))/count(NULLIF(robbery, 0)), 
                                         sum(NULLIF(sexual_violence_minors, 0))/count(NULLIF(sexual_violence_minors, 0)), sum(NULLIF(sexual_violence_all, 0))/count(NULLIF(sexual_violence_all, 0))
                                         FROM data
                                         WHERE spirit >= {spirit_avg}'''.format(spirit_avg = spirit_avg)).fetchall()
sca = spirit_crime_ave
wine_crime_ave = con.execute('''SELECT sum(NULLIF(assault, 0))/count(NULLIF(assault, 0)), sum(NULLIF(burglary, 0))/count(NULLIF(burglary, 0)), sum(NULLIF(burglary_domestic, 0))/count(NULLIF(burglary_domestic, 0)), 
                                       sum(NULLIF(homicide, 0))/count(NULLIF(homicide, 0)), sum(NULLIF(motor_vehicle_theft, 0))/count(NULLIF(motor_vehicle_theft, 0)), sum(NULLIF(robbery, 0))/count(NULLIF(robbery, 0)), 
                                       sum(NULLIF(sexual_violence_minors, 0))/count(NULLIF(sexual_violence_minors, 0)), sum(NULLIF(sexual_violence_all, 0))/count(NULLIF(sexual_violence_all, 0))
                                       FROM data
                                       WHERE wine >= {wine_avg}'''.format(wine_avg = wine_avg)).fetchall()
wca = wine_crime_ave

display(HTML("<table>  <tr> <th> Crime </th>                       <th> At or Above Average Beer Rate </th>  <th> At or Above Average Spirit Rate </th> <th> At or Above Average Wine Rate </th>   <th> World Average </th> </tr> \
                       <tr> <td> Assault </td>                     <td> {b1} </td>                           <td> {s1} </td>                            <td> {w1} </td>                            <td> 220.33 </td> </tr> \
                       <tr> <td> Burglary </td>                    <td> {b2} </td>                           <td> {s2} </td>                            <td> {w2} </td>                            <td> 341.71 </td> </tr> \
                       <tr> <td> Burglary Domestic </td>           <td> {b3} </td>                           <td> {s3} </td>                            <td> {w3} </td>                            <td> 166.93 </td> </tr> \
                       <tr> <td> Homicide </td>                    <td> {b4} </td>                           <td> {s4} </td>                            <td> {w4} </td>                            <td> 8.01 </td>   </tr> \
                       <tr> <td> Motor Vehicle Theft </td>         <td> {b5} </td>                           <td> {s5} </td>                            <td> {w5} </td>                            <td> 92.39 </td>  </tr> \
                       <tr> <td> Robbery </td>                     <td> {b6} </td>                           <td> {s6} </td>                            <td> {w6} </td>                            <td> 113.97 </td> </tr> \
                       <tr> <td> Sexual Violence (Minors) </td>    <td> {b7} </td>                           <td> {s7} </td>                            <td> {w7} </td>                            <td> 12.87 </td>  </tr> \
                       <tr> <td> Sexual Violence (All) </td>       <td> {b8} </td>                           <td> {s8} </td>                            <td> {w8} </td>                            <td> 31.04 </td>  </tr> \
                       </table>".format(b1 = round(bca[0][0],2), b2 = round(bca[0][1],2), b3 = round(bca[0][2],2), b4 = round(bca[0][3],2), b5 = round(bca[0][4],2), b6 = round(bca[0][5],2), b7 = round(bca[0][6],2), b8 = round(bca[0][7],2),
                                        s1 = round(sca[0][0],2), s2 = round(sca[0][1],2), s3 = round(sca[0][2],2), s4 = round(sca[0][3],2), s5 = round(sca[0][4],2), s6 = round(sca[0][5],2), s7 = round(sca[0][6],2), s8 = round(sca[0][7],2),
                                        w1 = round(wca[0][0],2), w2 = round(wca[0][1],2), w3 = round(wca[0][2],2), w4 = round(wca[0][3],2), w5 = round(wca[0][4],2), w6 = round(wca[0][5],2), w7 = round(wca[0][6],2), w8 = round(wca[0][7],2))))

Countries that consume an above average amount of beer are more likely to be involved in: assault, burglary, domestic burglary, motor vehicle theft, robbery, and both types of sexual violence. Countries that consume an above average amount of spirits are more likely to be involved in: assault, domestic burglary, homicide, and sexual violence. Countries that consume an above average amount of wine are more likely to be involved in: burglary, domestic burglary, motor vehicle theft, robbery, and sexual violence. The general takeaway from this table is that consuming an above average amount of alcohol for a country is generally detrimental to countries crime rates. The most comical insight I noticed was the increase rate of assault given an over consumption of beer. As mentioned before, alcohol is generally associated with people being loud and over aggressive. Furthermore, college aged males drinking beer are even more associated with this stereotype. I have no proof, but I found it comical that this could be the explanation for the increase in assault rate.

# The United States Of America

In the final section I wanted to determine if alcohol can be a good predictor for criminal activity. Above I discovered that generally alcohol is a poor predictor for criminal activity. I wanted to challenge this idea and determine if I could accurately predict the criminal rates for the United States based soley on their alchol consumption. I used the total amount of alcoholc consumed by the USA to make a prediction regarding the crime statistics. I selected 11 countries (the list of countries can be seen below) that all have a total consumption value within 0.4 gallons of the US. I grouped all of their statistics for each crime and took the average. I then used this average to represent a prediction for the US rates. 

In [13]:
data[184]['country']
USA_beer = data[184]['beer_servings']
USA_spirit = data[184]['spirit_servings']
USA_wine = data[184]['wine_servings']
USA_total = data[184]['total_litres_of_pure_alcohol']

In [14]:
con.execute('''SELECT country
               FROM data
               where (pure >= ({USA_total} - .2) and pure <= ({USA_total} + .2))'''.format(USA_total = USA_total)).fetchall()

[('Argentina',),
 ('Canada',),
 ('Cyprus',),
 ('Gabon',),
 ('Greece',),
 ('Netherlands',),
 ('New Zealand',),
 ('Nigeria',),
 ('South Africa',),
 ('Uganda',),
 ('Ukraine',),
 ('USA',)]

In [15]:
assault_predict = round(con.execute('''SELECT sum(NULLIF(assault,0))/count(NULLIF(assault,0)) as average
                                 FROM data
                                 WHERE (country = 'Argentina' or country = 'Canada' or country = 'Cyprus' or country = 'Gabon' or country = 'Greece' or country = 'Netherlands'
                                 or country = 'New Zealand' or country = 'Nigeria' or country = 'South Africa' or country = 'Uganda' or country = 'Ukraine')''').fetchall()[0][0],2)

burglary_predict = round(con.execute('''SELECT sum(NULLIF(burglary,0))/count(NULLIF(burglary,0)) as average
                                  FROM data
                                  WHERE (country = 'Argentina' or country = 'Canada' or country = 'Cyprus' or country = 'Gabon' or country = 'Greece' or country = 'Netherlands'
                                  or country = 'New Zealand' or country = 'Nigeria' or country = 'South Africa' or country = 'Uganda' or country = 'Ukraine')''').fetchall()[0][0],2)

homicide_predict = round(con.execute('''SELECT sum(NULLIF(homicide,0))/count(NULLIF(homicide,0)) as average
                                  FROM data
                                  WHERE (country = 'Argentina' or country = 'Canada' or country = 'Cyprus' or country = 'Gabon' or country = 'Greece' or country = 'Netherlands'
                                  or country = 'New Zealand' or country = 'Nigeria' or country = 'South Africa' or country = 'Uganda' or country = 'Ukraine')''').fetchall()[0][0],2)

motor_predict = round(con.execute('''SELECT sum(NULLIF(motor_vehicle_theft,0))/count(NULLIF(motor_vehicle_theft,0)) as average
                               FROM data
                               WHERE (country = 'Argentina' or country = 'Canada' or country = 'Cyprus' or country = 'Gabon' or country = 'Greece' or country = 'Netherlands'
                               or country = 'New Zealand' or country = 'Nigeria' or country = 'South Africa' or country = 'Uganda' or country = 'Ukraine')''').fetchall()[0][0],2)

robbery_predict = round(con.execute('''SELECT sum(NULLIF(robbery,0))/count(NULLIF(robbery,0)) as average
                                 FROM data
                                 WHERE (country = 'Argentina' or country = 'Canada' or country = 'Cyprus' or country = 'Gabon' or country = 'Greece' or country = 'Netherlands'
                                 or country = 'New Zealand' or country = 'Nigeria' or country = 'South Africa' or country = 'Uganda' or country = 'Ukraine')''').fetchall()[0][0],2)

sexual_assault_predict = round(con.execute('''SELECT sum(NULLIF(sexual_violence_all,0))/count(NULLIF(sexual_violence_all,0)) as average
                                        FROM data
                                        WHERE (country = 'Argentina' or country = 'Canada' or country = 'Cyprus' or country = 'Gabon' or country = 'Greece' or country = 'Netherlands'
                                        or country = 'New Zealand' or country = 'Nigeria' or country = 'South Africa' or country = 'Uganda' or country = 'Ukraine')''').fetchall()[0][0],2)

In [16]:
display(HTML("<table>  <tr> <th> Crime </th> <th> World Average </th><th> Predicted Rate </th> <th> Actual Rate </th> <th> Percent Difference (Actual - Predicted)/Actual </th> </tr> \
                       <tr> <td> Assault </td> <td> 220.33 </td> <td> {assault_predict} </td> <td> 252.8 </td> <td> 36.08% </td> </tr> \
                       <tr> <td> Burglary </td> <td> 341.71 </td> <td> {burglary_predict} </td> <td> 701.0 </td> <td> 9.86% </td> </tr> \
                       <tr> <td> Homicide </td> <td> 8.01 </td> <td> {homicide_predict} </td> <td> 4.8 </td> <td> -64.79% </td> </tr> \
                       <tr> <td> Motor Vehicle Theft </td> <td> 92.39 </td> <td> {motor_predict} </td> <td> 239.1 </td> <td> 24.41% </td> </tr> \
                       <tr> <td> Robbery </td> <td> 113.97 </td> <td> {robbery_predict} </td> <td> 119.3 </td> <td> -35.62% </td> </tr> \
                       <tr> <td> Sexual Violence (All) </td> <td> 31.04 </td> <td> {sexual_assault_predict} </td> <td> 27.7 </td>  <td> -22.13% </td> </tr> \
                       </table>".format(assault_predict = assault_predict, burglary_predict = burglary_predict, homicide_predict = homicide_predict,
                                        motor_predict = motor_predict, robbery_predict = robbery_predict, sexual_assault_predict = sexual_assault_predict)))
print("The United States consumed", round(USA_total,2), "gallons of alcohol per person in 2010. This value was 0.96 gallons higher than the global average.")

The United States consumed 2.3 gallons of alcohol per person in 2010. This value was 0.96 gallons higher than the global average.


The queries confirmed what was put forth above, that alcohol consumption is not an accurate predictor for criminal activity. The predicted values averaged a 32.15% difference from the actual value. In order to improve the prediction you could include a wider range in order to base the prediction off of a greater number of countries. However, even this method would not correctly identify the criminal rates. This is due to the fact that alcohol consumption does not cause criminal activity and is a rather poor predictor of criminal activity. The United States is a prime example of this. When comparing the United States crime rates to the world averages it becomes apparent that the United States is an anomaly. The United States had a similar rate for assault, robbery, and sexual violence with the rest of the world. Yet, the United States had over twice the rate of burglary, over twice the rate of motor vehicle theft, and half of the homicide rate. This example shows the difficulty in attempting to predict criminal activity from alcohol consumption. These queries also highlighted how effective prediction from criminal activity to alcohol consumption can be. In the case of the United States it had elevated rates in four out of the six crimes suggesting that the United States consumed more alcohol than the global average in 2010. This fact is true: the United States consumed 2.3 gallons per person in comparison to the world average of 1.32 gallons per person.  

# Conclusion

This report outlined the creation of database consisting of alcohol consumption and criminal records. The report attempted to determine if there are any correlations between alcohol consumption and criminal activity. It was found that criminal activity can be a fairly accurate predictor for alcohol consumption. Knowing the rates of crime within a country can be used along with other methods to correctly identify if a country consumes an above average amount of alcohol. The inverse of this relationship was found to be faulty. Alcohol consumption is a very poor predictor of criminal activity. This discovery highlighted the idea that criminal activity is present in every country no matter the amount of alcohol consumed. It was also discovered that different types of alcohol are capable of implying criminal activity. These types of alcohol are not perfect predictors, but they are capable of serving as potential predictors. Overall, the report highlights that there is a correlation between criminal activity and alcohol consumption.